neurolib/models/multimodel/builder/adex.py

```diff
@@ -0,0 +1,944 @@
+            + self.params["ext_current"]
+            - I_mu
+        ) / tau
+
+        d_I_adaptation = (
+            self.params["a"] * (voltage - self.params["E_A"])
+            - I_adaptation
+            + self.params["tau_A"] * self.params["b"] * firing_rate_now
+        ) / self.params["tau_A"]
+
+        d_I_syn_mu_exc = ((1.0 - I_syn_mu_exc) * exc_inp - I_syn_mu_exc) / self.params["tau_syn_exc"]
+
+        d_I_syn_mu_inh = ((1.0 - I_syn_mu_inh) * inh_inp - I_syn_mu_inh) / self.params["tau_syn_inh"]
+
+        d_I_syn_sigma_exc = (
+            (1.0 - I_syn_mu_exc) ** 2 * exc_inp_sq
+            + (exc_inp_sq - 2.0 * self.params["tau_syn_exc"] * (exc_inp + 1.0)) * I_syn_sigma_exc
+        ) / (self.params["tau_syn_exc"] ** 2)
+
+        d_I_syn_sigma_inh = (
+            (1.0 - I_syn_mu_inh) ** 2 * inh_inp_sq
+            + (inh_inp_sq - 2.0 * self.params["tau_syn_inh"] * (inh_inp + 1.0)) * I_syn_sigma_inh
+        ) / (self.params["tau_syn_inh"] ** 2)
+        # firing rate as dummy dynamical variable with infinitely fast
+        # fixed-point dynamics
+        d_firing_rate = -self.params["lambda"] * (firing_rate - firing_rate_now)
+
+        return [
+            d_I_mu,
+            d_I_adaptation,
+            d_I_syn_mu_exc,
+            d_I_syn_mu_inh,
+            d_I_syn_sigma_exc,
+            d_I_syn_sigma_inh,
+            d_firing_rate,
+        ]
+
+
+class InhibitoryAdExMass(AdExMass):
+    """
+    Inhibitory AdEx neural mass. In contrast to excitatory, inhibitory mass do
+    not contain fiting rate adaptation current.
+    """
+
+    name = "AdEx mean-field inhibitory mass"
+    label = f"AdExMass{INH}"
+    num_state_variables = 6
+    coupling_variables = {5: f"q_mean_{INH}"}
+    mass_type = INH
+    state_variable_names = [
+        "I_mu",
+        "I_syn_mu_exc",
+        "I_syn_mu_inh",
+        "I_syn_sigma_exc",
+        "I_syn_sigma_inh",
+        "q_mean",
+    ]
+    required_couplings = [
+        "node_inh_exc",
+        "node_inh_exc_sq",
+        "node_inh_inh",
+        "node_inh_inh_sq",
+    ]
+    required_params = [
+        "K_exc",
+        "K_inh",
+        "c_global",
+        "K_exc_global",
+        "tau_syn_exc",
+        "tau_syn_inh",
+        "sigma_ext",
+        "J_exc_max",
+        "J_inh_max",
+        "tau_m",
+        "C_m",
+        "ext_current",
+        "ext_rate",
+        "lambda",
+    ]
+
+    def __init__(self, params=None, lin_nonlin_cascade_filename=None, seed=None):
+        super().__init__(
+            params=params or DEFAULT_PARAMS_INH, lin_nonlin_cascade_filename=lin_nonlin_cascade_filename, seed=seed
+        )
+
+    def _initialize_state_vector(self):
+        """
+        Initialize state vector.
+        """
+        np.random.seed(self.seed)
+        self.initial_state = (
+            np.random.uniform(0, 1, self.num_state_variables) * np.array([3.0, 0.5, 0.5, 0.01, 0.01, 0.01])
+        ).tolist()
+
+    def _compute_couplings(self, coupling_variables):
```

Comment:
dynamics definition for AdEx is kinda ugly and long so I defined some helper convenience functions such as this - this just computes coupling - both inh -> inh and exc-> inh and their squared variants